neevaco/neeva-android
# Fix previews and prevent sending feedback without text

* Make `SessionToken` use `by lazy` to instantiate some of its fields instead of directly instantiating it.  This is better in the long run because it means that we're not creating objects we don't need immediately, but necessary to make `@Preview`s work again because the Previews don't know how to instantiate the Moshi or OkHttpClient objects.

* Disable the feedback sending button until the user has typed in some text.

## Checklists

### How was this tested?

### Manual test cases
See the videos: tested the `FullScreenDialogTopBar` on the two places that use it with a button.

### Screenshots and screen recordings

https://user-images.githubusercontent.com/96092902/192399334-62d75962-c571-4ad9-9063-dcf5770592e6.mp4

https://user-images.githubusercontent.com/96092902/192401481-0f900c0e-a874-4680-a605-68c4a8bf7aea.mp4

<img width="1762" alt="Screen Shot 2022-09-26 at 4 34 14 PM" src="https://user-images.githubusercontent.com/96092902/192399342-8e630ff9-d31b-4f18-acc9-0ebfc0ef37b0.png">

## Issues addressed
* Fixes https://github.com/neevaco/neeva-android/issues/989
